### PR TITLE
Connect sign in with backend / Load nodes

### PR DESCRIPTION
**Pull request by irubido**
Changes requested from Load nodes branch

1.
> handle case if no nodes

2.
> You can write a helper function that will return headers config. Then you won't have repeated code regarding authorization token and if the authorization key needs to change at some point, we can change it on one place only.
> 
> Example:
> 
> const response = await axios.get(url, {
>    ...getAuthHeaders(auth),
>    params: {...}
> }

### CODE REVIEW

File: src/app/rootReducer.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import {combineReducers} from "@reduxjs/toolkit";
 import registerReducer from "../containers/Register/RegisterSlice";
+import loginReducer from "../containers/Login/LoginSlice";
 
 const rootReducer = combineReducers({
     register: registerReducer,
+    login: loginReducer,
```
Comment (morrigan) [inline src/app/rootReducer.ts]: maybe it's better that we rename to `user` reducer and actions only and put there everything

File: src/containers/Dashboard/DashboardContainer.tsx [hunk 2]
```diff
@@ -10,10 +9,20 @@ import {DiskSpace} from "../DiskSpace/DiskSpaceContainer";
...
+import {useSelector} from "react-redux";
+import {RootState} from "../../app/rootReducer";
+import {useHistory} from "react-router-dom";
+import {Routes} from "../../constants/routes";
 
 export const DashboardContainer = (): ReactElement => {
     const [areElementsHidden, setElementsHidden] = useState(false);
+    const history = useHistory();
+    const stateToken = useSelector((state: RootState) => state.login.token);
+    const storageToken = localStorage.getItem("token");
```
Comment (morrigan) [inline src/containers/Dashboard/DashboardContainer.tsx]: This should definitely not be here. You can have an initial state loading from local storage.

File: src/containers/Dashboard/DashboardContainer.tsx [hunk 3]
```diff
@@ -13,14 +13,23 @@ import {useSelector} from "react-redux";
...
 import {useHistory} from "react-router-dom";
 import {Routes} from "../../constants/routes";
+import jwt_decode from "jwt-decode";
 
 export const DashboardContainer = (): ReactElement => {
     const [areElementsHidden, setElementsHidden] = useState(false);
     const history = useHistory();
     const stateToken = useSelector((state: RootState) => state.user.token);
 
+    const checkTokenExpireTime = (token: string | null): boolean => {
```
Comment (morrigan) [inline src/containers/Dashboard/DashboardContainer.tsx]: This is kinda common for all pages in the app (even though we have dashboard now only) so would be good if you put it behind in the service, redux state logic. For example, you could move this function when initializing state with token. If it is expired set token in state as null. Then the authenticated routes logic that you wrote works automatically.

File: src/containers/Dashboard/DashboardApi.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import {config} from "../../app/config";
+
+export async function getNodes(auth: string | null) {
+    const url = `${config.apiURL}/user/node`;
+
+    try {
+        const response = await axios.get(url, {
+            headers: {
```
Comment (morrigan) [inline src/containers/Dashboard/DashboardApi.ts]: I don't see that you have resolve the comment regarding headers?

File: src/containers/Dashboard/NodeSlice.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+export const {loading, storeNodeList, storeNodeInfo} = nodeSlice.actions;
+export default nodeSlice.reducer;
+
+export const getNodeList = (auth: string | null): AppThunk => async dispatch => {
+    try {
+        const nodeListResponse = await getNodes(auth);
+        dispatch(storeNodeList(nodeListResponse.data));
+
+        if (nodeListResponse.data[0] && nodeListResponse.data[0].id) {
+            const nodeInfoResponse = await getMinerInfo(auth, nodeListResponse.data[0].id);
```
Comment (morrigan) [inline src/containers/Dashboard/NodeSlice.ts]: Functions should have single responsabilities. Meaning, you can't have fetching node list and then getting miner info in the same function. You can leave for now but this logic has to change when resolving #14 anyway

File: src/containers/GeneralInfo/GeneralInfo.tsx [hunk 6]
```diff
@@ -1,13 +1,17 @@
...
+import {useSelector} from "react-redux";
+import {RootState} from "../../app/rootReducer";
 
 interface IGeneralInfoProps {
     setElementsHidden: Dispatch<SetStateAction<boolean>>;
     areElementsHidden: boolean;
 }
 
 export const GeneralInfo = ({setElementsHidden, areElementsHidden}: IGeneralInfoProps): ReactElement => {
+    const node = useSelector((state: RootState) => state.node);
```
Comment (morrigan) [inline src/containers/GeneralInfo/GeneralInfo.tsx]: I don't see how you handled the case if there is no node (node list is empty)? I can add a separate issue for that then with more UI details.

However, have in mind those edge cases.

File: src/containers/Dashboard/NodeSlice.ts (hunk 5 above)
Comment (irubido) [reply src/containers/Dashboard/NodeSlice.ts]: Yes now all fetch functions are in the same dispatch getNodeList, I will take 
#14 next and rearrange the logic and implement switching nodes.

File: src/containers/GeneralInfo/GeneralInfo.tsx (hunk 6 above)
Comment (irubido) [reply src/containers/GeneralInfo/GeneralInfo.tsx]: I handle it on new commit. 
App loads node list page by default(flow from the figma design), if nodeList is fetched, app shows statistics for the 1. one in the array.

If there are no nodes, arrow to show statistics is hidden and app displays "EmptyList" component.

**Pull request merged**